### PR TITLE
Refactor document properties overlay

### PR DESCRIPTION
**Pull request by timvandermeij**
This PR follows a similar pattern as other refactoring patches. The aim here is to make the document properties code more class-like, simplifying field passing/handling and avoiding passing a reference to the secondary toolbar.

_Using the_ `?w=1` _URL parameter makes review easier._

### CODE REVIEW

**Comment by timvandermeij** [comment]
@Snuffleupagus Thank you for the thorough review! I will make the changes in a new set of commits.

**Comment by timvandermeij** [comment]
@Snuffleupagus The nits have been addresses. No functional changes have been made.

File: web/pdf_document_properties.js [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+  function PDFDocumentProperties(options) {
+    this.fields = options.fields;
+    this.overlayName = options.overlayName;
+
+    this.rawFileSize = 0;
+    this.url = null;
+    this.pdfDocument = null;
+
+    // Bind the event listener for the Close button.
+    if (options.closeButton) {
```
Comment (Snuffleupagus) [inline web/pdf_document_properties.js]: Sorry about not catching this earlier, but the button is missing from the JSDoc comment above. `{HTMLButtonElement}` would probably be an OK type to use.

The PR is good to go once this is addressed.

Comment (timvandermeij) [inline web/pdf_document_properties.js]: Addressed in the most recent commit.

**Comment by Snuffleupagus** [comment]
Looks great, thank you!

**Pull request merged**